Add Bust-aligned buttons option to Simone. Resolves #2154

BenJamesBen opened the pull request:
This PR adds a Bust-aligned Buttons option to Simone, with 3 settings: Disabled (default), Even spacing, Split spacing. By default, the existing button spacing is used. With the setting set to Even or Split spacing, button spacing is arranged to ensure that a button is placed at the bustline to prevent possible gaping. This fulfills Enhancement Request #2154 .

The PR also:
- Adds a notch at the center front bustline to help sewists who might want to add a snap or other closure there, even if bust-aligned button spacing isn't used.
- Changes the location of points.bust after the FBA so it is the actual bust point. Additional comments are added to the code, documenting this change.
- Fixes the center front hem notch on separate buttonhole placket by moving it to the correct adjusted location.
- Re-adds the center front armhole, waist, and hem notches to the button placket. These notches were accidentally removed in PR #2477 .
- Removes a superfluous bust point notch from the buttonhole placket (overlooked in PR #2477 ).
- Updates changelog files.

I believe that I tested the changes to make sure they work with all designs and styles: with separate and attached button/buttonhole packets, with classic and seamless button/buttonhole plackets, and with all equivalent Simon variations to check for regressions.

joostdecock reviewed (changes requested):
Excellent PR 🙌 

If you move the option under *style* => *closure* in the option groups, I will merge it.
If you are unsure how to do so, let me know and I will merge it, and then add it.

Thanks for another awesome contribution 🙏

BenJamesBen commented on designs/simone/config/index.js:
```diff
@@ -7,7 +7,12 @@ const config = {
   optionGroups: {
     ...simonConfig.optionGroups,
-    style: [...simonConfig.optionGroups.style, 'frontDarts', 'contour'],
+    style: [
```
I moved the Bust-Aligned Buttons option into the Style->Closure submenu.

BenJamesBen commented:
If there was a more straightforward method I could/should have used to add the Bust-aligned Buttons option into the existing Closures submenu, please let me know.

joostdecock commented on designs/simone/config/index.js:
```diff
@@ -1,6 +1,24 @@
+//   array. This seems to be the least disruptive method.
+const closureindex = simonConfig.optionGroups.style.findIndex((item) =>
+    typeof item === 'object' && "closure" in item)
+simonConfig.optionGroups.style[closureindex].closure.push('bustAlignedButtons')
```
Nicely done. FYI, we're going to change this in v3 to make it easier to handle this.
We'll only accept a top-level key and an array of options. For nested structure, you would use a `:` in they key.

So here you would have just added:
```js
"style:closure": [ "bustAlignedButtons"]
```
And that would then get merged into the closure group.

Still on the fence about using `:` or `.` as a separator, but this is the idea.

joostdecock commented on packages/i18n/src/locales/en/options/simone.yml:
```diff
@@ -1,3 +1,7 @@
+bustAlignedButtons:
```
If we move this to `simon.yml` the tests should pass (and it will get inherited by simone anyway)

BenJamesBen replied:
I moved the option translation from Simone to Simon.

joostdecock commented:
> Also please let me know if I need to add anything to Simon

Since you've added the option, you should add the translation to simon too, that will fix the tests.
And as simone inherits its translations from simon, it should *just work*.

BenJamesBen commented:
I removed the Bust-aligned Buttons option and translation from Simon and moved them entirely to Simone. Trying to put them into Simon was the wrong thing to do, to work around an error. I changed simon/src/shared.js to fix the error (to simply check whether the bustAlignedButtons option existed before trying to use it).

I also fixed the translation to have the list options translated as well, which was the way I should have done it in the first place.

I made changes to:
```
	modified:   designs/simon/config/index.js
	modified:   designs/simon/src/shared.js
	modified:   designs/simone/config/index.js
	modified:   packages/i18n/src/locales/en/options/simon.yml
	modified:   packages/i18n/src/locales/en/options/simone.yml
```

I am still getting an i18n test error in Simon, and I am investigating why this is occurring. Simon should no longer be aware of the bustAlignedButtons option, so it should not expect a translation for it:
`Error: No option translation found for bustAlignedButtons in simon`
